File: Go/min_cost_climbing_stairs.go
func minCostClimbingStairs(cost []int) int {
    if len(cost) == 0 {return 0}
    if len(cost) == 1 {return cost[0]}
    a, b := cost[0], cost[1]
    for i := 2; i < len(cost); i++ {
        a, b = b, min(a, b) + cost[i]
    }
    return min(a, b)
}

func min(a, b int) int {
    if a < b {return a}
    return b
}

File: Java/MinCostClimbingStairs.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        if(cost == null || cost.length == 0) return 0;
        if(cost.length == 1) return cost[0];

        int a = cost[0], b = cost[1];
        for(int i = 2; i < cost.length; i++) {
            int cur = Math.min(a, b) + cost[i];
            a = b;
            b = cur;
        }
        return Math.min(a, b);
    }
}